File: 2015/01/part1.ex
defmodule AOC2015.Day1 do

  def solve(str) do
    str |> String.graphemes |> Enum.map(&brace_to_int/1) |> Enum.sum
  end

  defp brace_to_int(x) when x == "(" do 1 end
  defp brace_to_int(_) do  -1 end

end

{ :ok, str } = File.read('./input.txt')
IO.puts AdventOfCode.Day1.solve(str)

File: 2017/01/part1.rb
input = File.read('./input.txt')

sum = 0
handler = ->(a, b) { sum += a.to_i if a == b }

handler.call(input[0], input[-1])
input.chars.each_cons(2) { |a, b| handler.call(a, b) }

puts sum